### DISCUSSION TITLE
CTkScrollableFrame- Are double scrollbars possible?

### DISCUSSION THREAD

**Discussion by Eloque** [TomSchimansky/CustomTkinter#2215] [Q&A] [answered]
I am building an app to control a laser cutter, and the bed is represented by a canvas on a CTkScrollableFrame. 
As the user can zoom in and out, it would be help full if this frame was scrollable in both directions. Is that possible?

Looking at the code I see this
```
        if self._orientation == "horizontal":
            self._scrollbar = CTkScrollbar(master=self._parent_frame, orientation="horizontal", command=self._parent_canvas.xview,
                                           fg_color=scrollbar_fg_color, button_color=scrollbar_button_color, button_hover_color=scrollbar_button_hover_color)
            self._parent_canvas.configure(xscrollcommand=self._scrollbar.set)
        elif self._orientation == "vertical":
            self._scrollbar = CTkScrollbar(master=self._parent_frame, orientation="vertical", command=self._parent_canvas.yview,
                                           fg_color=scrollbar_fg_color, button_color=scrollbar_button_color, button_hover_color=scrollbar_button_hover_color)
            self._parent_canvas.configure(yscrollcommand=self._scrollbar.set)
```

So I am assuming it is not possible out of the box.

**Comment by Akascape** [comment, marked as answer]
@Eloque Possible, but you have to use this external package: https://github.com/Akascape/CTkXYFrame

![Screenshot](https://github.com/Akascape/CTkXYFrame/assets/89206401/859b6733-3c91-4093-a511-d6f1060a18ca)

**Comment by Eloque** [comment]
That looks really good. I am going to try that thank you. I like Python, but I never really clicked with any gui framework for it.
Also, that is an impressive Github account you got there.

**Reply by Akascape** [reply]
@Eloque Thanks, you can find more gui elements here: https://github.com/Akascape/tkinter-toolkit

**Reply by Eloque** [reply]
Looks cool, and it does seem to solve my problem. I was also looking for a file picker, so that works out.
One thing I need to figure out is how to get the width of the parent frame, as the canvas needs to fill the canvas out to the max size and caculate the correct scale factor at startup

```
        # Get the current width of the canvas parent frame
        width = self.main_frame.winfo_width()

        # Set the width of the canvas to the width of the parent frame
        self.canvas.configure(width=width)
```

Keeps giving the same size, regardless of the resizing I do.

```
width = self.main_frame.parent_frame.winfo_width()
```

Works, but gives the full size, including the scrollbar, so probably have to adjust for that.
